fix(agora): dialog, misc styling issues (AG-1751)

sagely1 opened the pull request:
### Description
This addresses styling issues in dialogs and icons.

GCT overlay header:
Before:
<img width="602" alt="image" src="https://github.com/user-attachments/assets/f3c4a6c2-744c-40d8-a9f4-3f94ce98542b" />

After:
<img width="602" alt="image" src="https://github.com/user-attachments/assets/1df3f2e7-daef-4e9a-957a-a95d90066186" />

GCT overlay icon
Before:
<img width="995" alt="image" src="https://github.com/user-attachments/assets/f1ec6810-faea-4266-bd16-050ccde17b79" />

After:
<img width="841" alt="image" src="https://github.com/user-attachments/assets/2924ea25-81f4-4146-9ac4-55a3ee053d7d" />

Legend
Before (Production Agora):
This addresses an existing bug in production with the "Color" heading missing:
<img width="604" alt="image" src="https://github.com/user-attachments/assets/bc4fab57-6dd9-4c0a-9473-5f25c4bbfb55" />

After:
<img width="604" alt="image" src="https://github.com/user-attachments/assets/bfb00cf2-0784-4b9d-bcee-a079b848ecf8" />

Nominated Targets "Learn more about Nominated Targets" link
Before:
<img width="986" alt="image" src="https://github.com/user-attachments/assets/35a19305-ca81-42a3-ad59-3e9316da5ac1" />

After:
<img width="981" alt="image" src="https://github.com/user-attachments/assets/bfd2b47f-ad73-49c1-990c-d21d238816a6" />

Gene Details - Evidence - RNA: Data and Methods for this study link
Before:
<img width="991" alt="image" src="https://github.com/user-attachments/assets/ec6e449e-91a8-4b4b-8a60-b79908b84e3d" />

After:
<img width="989" alt="image" src="https://github.com/user-attachments/assets/ea8fb53a-9046-47f6-878e-8d4748c476f2" />

Gene Details - Evidence - RNA: similar link
Before:
<img width="45" alt="image" src="https://github.com/user-attachments/assets/3a4f9f23-9d5f-43ba-b39d-27626b4da476" />

After:
<img width="37" alt="image" src="https://github.com/user-attachments/assets/f8cce0b6-e4e9-4676-99b0-f66e51d3a7f6" />

hallieswan reviewed (changes requested):
The updates to the `popover-link` dialog and `gene-similar` link icon look great! 

For the `p-dialog` styling, it looks like we're hiding the PrimeNG close button and adding a custom close button in the header of each dialog, then applying similar styling in each component. Instead, could we style the PrimeNG close button and header globally or in a shared place? That way, the dialogs will stay consistent going forward and we can avoid duplicating code. 

I think we could use the existing `_dialog.scss` if we want (and tackle moving the styles into the preset in the future) -- we would just need to update class names that were changed in the PrimeNG update (`.p-dialog-header-icons` --> `.p-dialog-header-actions`, `.p-dialog-header-close-icon` --> `.p-dialog-close-button`) and tweak a couple global styles in the preset (`dialog.header.padding: '0'`, `dialog.root.borderRadius: '0'`).

sagely1 commented:
Good point.  I wanted to duplicate the existing visual behavior so I added the ability to customize the dialog per component but this makes me feel like the right way forward is to stick with a consistent styling.  Since I wanted to clean up the shared styles, I only brought back what was actually needed and reverted PrimeNG preset changes.  Now dialogs will be using a PrimeNG button (in the header) and not the custom SVGIconComponent so any styling or customization will have to occur in modal-link component going forward.  Another side-effect is that PrimeNG buttons have a slightly different hover behavior. 
 This means buttons in the dialogs will now have a circular background around the icon which is more visually apparent the icon is clickable.  I'm personally ok with this behavior.  
<img width="92" alt="image" src="https://github.com/user-attachments/assets/d5e6e8ce-d71a-4a63-9b70-b07238543d95" />

JessterB commented:
@hallieswan This actually fixes https://sagebionetworks.jira.com/browse/AG-1752, so yes please!
